source/client/process_impl.cc

```diff
@@ -260,11 +262,31 @@ void ProcessImpl::createBootstrapConfiguration(envoy::config::bootstrap::v2::Boo
         options_.maxPendingRequests() == 0 ? 1 : options_.maxPendingRequests());
     thresholds->mutable_max_requests()->set_value(options_.maxActiveRequests());
 
-    cluster->set_type(envoy::api::v2::Cluster::DiscoveryType::Cluster_DiscoveryType_STATIC);
-    auto* host = cluster->add_hosts();
-    auto* socket_address = host->mutable_socket_address();
-    socket_address->set_address(uri.address()->ip()->addressAsString());
-    socket_address->set_port_value(uri.port());
+    if (options_.backendEndpoints().empty()) {
+      cluster->set_type(envoy::api::v2::Cluster::DiscoveryType::Cluster_DiscoveryType_STATIC);
+      auto* host = cluster->add_hosts();
+      auto* socket_address = host->mutable_socket_address();
+      socket_address->set_address(uri.address()->ip()->addressAsString());
+      socket_address->set_port_value(uri.port());
+    } else {
+      if (Utility::hostAddressTypeFromHostPort(options_.backendEndpoints()[0]) ==
+          HostAddressType::DNS) {
+        cluster->set_type(envoy::api::v2::Cluster::DiscoveryType::Cluster_DiscoveryType_STRICT_DNS);
```

Comment:
There's probably something I'm missing here, but why is this possible now, but wasn't before?  If this is a new feature (allowing Strict DNS), should we allow that for uri above as well?

Reply:
Good point, in URI mode it was resolving the DNS to an IP address ahead of time, according to an auto/v4/v6 option. I now updated --multi-target-* mode to uniformly reuse the same resolution code, so this DNS-related code no longer exists.